libethash-cl/CLMiner.cpp

```diff
@@ -101,8 +101,8 @@ void CLMiner::report(uint64_t _nonce, WorkPackage const& _w)
 {
 	assert(_nonce != 0);
 	// TODO: Why re-evaluating?
-	Result r = EthashAux::eval(_w.seed, _w.header, _nonce);
-	if (r.value < _w.boundary)
+	Result r;
+	if (!g_checkResults || ((r = EthashAux::eval(_w.seed, _w.header, _nonce)).value < _w.boundary))
```

Comment:
This is wrong, you are submitting empty "mix hash". If you want to avoid evaluation on CPU you have to send the mix hash back from GPU.

Reply:
true. It seems CUDA-version follows the same validate-on-CPU-logic:
https://github.com/ethereum-mining/ethminer/blob/master/libethash-cuda/CUDAMiner.cpp#L104

@jean-m-cyr The easiest way might be to ditch changes in CLMiner.cpp :( , because CUDAMIner.cpp has same kind of check logic.

Reply:
@chfast I know. Figured that out last night.

@androidi I'm thinking of ditching the whole thing, at least temporarily.

Had some strange occurrence overnight with this code. For a period of about 10 minutes I did issue a few invalid shares along with reporting bogus hash rates at the same time. One data point showed about twice the possible hashrate the the next showed half.

This tells me there's something fundamentally broken and I don't think it has anything to do with these changes. I'm going back to the latest master to let this latest image soak for a while.

Reply:
It's just the cumulated value that's oscillating. I've observed same kind of fluctuations, which are mostly harmless. There's a function call which accumulates the calculated number of hashes:
https://github.com/ethereum-mining/ethminer/blob/master/libethash-cuda/CUDAMiner.cpp#L73
(that's CUDA version, but OpenCL is pretty much the same)

If your batch size is large enough, there's a possibility it doesn't make it to the next flush (GPU still processing). Thus, sometimes you're getting higher-than-should-be values and lower-than-should-be values.

I find this "fluctuation" harmless; but it could/should preferably be some average over a bit longer period.